pallets/automation-time/src/migrations.rs

```diff
@@ -23,3 +23,41 @@ pub mod v1 {
 		T::DbWeight::get().reads_writes(1, 1)
 	}
 }
+
+pub mod v2 {
+	use frame_support::{migration::{storage_key_iter, get_storage_value}, traits::StorageVersion, Twox64Concat, BoundedVec};
+
+	use crate::{LastTimeSlot, Pallet, Task, Vec, TaskQueue, MissedQueue};
+
+	use super::*;
+
+	pub fn migrate<T: Config>() -> Weight {
+		info!(target: "automation-time", "Migrating automation-time v2");
+		let pallet_prefix: &[u8] = b"AutomationTime";
+
+		let scheduled_tasks_prefix: &[u8] = b"ScheduledTasks";
+		let _scheduled_tasks: Vec<_> = storage_key_iter::<u64, BoundedVec<T::Hash, T::MaxTasksPerSlot>, Twox64Concat>(pallet_prefix, scheduled_tasks_prefix)
```

Comment:
how does this work

Reply:
the .drain() is a function on an iterator, which removes everything from the storage map  https://docs.substrate.io/rustdocs//latest/src/frame_support/storage/migration.rs.html#61

and storage_key_iter just provides the iterator to drain the scheduledTasks storage: https://docs.substrate.io/rustdocs//latest/src/frame_support/storage/migration.rs.html#193

was following moonbeam here: https://github.com/PureStake/moonbeam/blob/e7b414a1370f7af5705e31835336460181be9267/pallets/asset-manager/src/migrations.rs#L82